addons/process-svg/process-svg-common/src/main/java/org/kie/kogito/svg/ProcessSvgService.java

```diff
@@ -0,0 +1,144 @@
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.ext.web.client.HttpResponse;
+import io.vertx.mutiny.ext.web.client.WebClient;
+
+import static java.util.Objects.isNull;
+
+import org.jbpm.process.svg.SVGImageProcessor;
+import org.jbpm.process.svg.processor.SVGProcessor;
+
+
+public abstract class ProcessSvgService {
+    public static final int RESPONSE_STATUS_CODE_OK = 200;
+
+    protected WebClient dataIndexWebClient;
+    protected WebClient addonWebClient;
+
+    protected String dataIndexHttpURL;
+    protected String svgResourcesPath;
+    protected String completedColor;
+    protected String completedBorderColor;
+    protected String activeBorderColor;
+
+    public String getSvgResourcesPath() {
+        return svgResourcesPath;
+    }
+
+    public WebClient getDataIndexWebClient() {
+        return dataIndexWebClient;
+    }
+
+    public void setDataIndexWebClient(WebClient dataIndexWebClient) {
+        this.dataIndexWebClient = dataIndexWebClient;
+    }
+
+    public WebClient getAddonWebClient(RoutingContext context) {
+        if(addonWebClient == null ){
+            setAddonWebClient(WebClient.create(Vertx.vertx(),  getWebClientToURLOptions(context.request().absoluteURI())));
+        }
+        return addonWebClient;
+    }
+
+    public void setAddonWebClient(WebClient addonWebClient) {
+        this.addonWebClient = addonWebClient;
+    }
+
+    public Uni<HttpResponse<Buffer>> getNodesQueryUni(String processId, String processInstanceId) {
+        String query = "{ ProcessInstances ( where: { and : {  id: {  equal : \"" + processInstanceId + "\" }, processId : { equal : \"" + processId + "\"} } }) { nodes { definitionId exit } } }";
+        return getDataIndexWebClient().post("/graphql")
+                .sendJson(JsonObject.mapFrom(Collections.singletonMap("query", query)));
+    }
+
+    public Uni<String> getSvgUni(String processId, RoutingContext context) {
+        return getAddonWebClient(context).get("/diagram/" + processId + ".svg")
+                .send()
+                .map(resp -> getSvgContent(resp));
+    }
+
+    protected String getSvgContent(HttpResponse<Buffer> resp) {
+        if (resp != null &&
+                resp.bodyAsString() != null &&
+                !resp.bodyAsString().contains("<title>404 - Resource Not Found</title>")) {
+            return resp.bodyAsString();
+        }
+        return "";
+    }
+
+    public String transformSvgToShowExecutedPath(String svg, List<String> completedNodes, List<String> activeNodes)  {
+        if (svg != null && !svg.isEmpty()) {
+            if (!(completedNodes.isEmpty() && activeNodes.isEmpty())) {
+                InputStream svgStream = new ByteArrayInputStream(svg.getBytes());
```

Comment:
Just make sure resources are closed
```suggestion
                try(InputStream svgStream = new ByteArrayInputStream(svg.getBytes())){
```